### PR TITLE
Feature/cbl2.5 replication eventing

### PR DESCRIPTION
**Pull request by hrajput89**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- Tests for Replication Eventing for both SG2DB and P2P for all platforms

### CODE REVIEW

File: CBLClient/Apps/CBLTestServer-iOS/Frameworks/CouchbaseLiteSwift.framework/Headers/CBLErrors.h [hunk 1]
```diff
@@ -0,0 +1,101 @@
+//
```
Comment (sridevi-15) [inline CBLClient/Apps/CBLTestServer-iOS/Frameworks/CouchbaseLiteSwift.framework/Headers/CBLErrors.h]: avoid checking in this file. remove this file from this commit

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+    sg_docs = sg_client.get_all_docs(url=sg_url, db=sg_db, auth=auth_session)
+    sg_docs = sg_docs["rows"]
+    sg_client.update_docs(url=sg_url, db=sg_db, docs=sg_docs, number_updates=1, auth=auth_session)
+
+    # Adding conflict for docs in CBL
+    db.update_bulk_docs(database=cbl_db, number_of_updates=1,
+                        doc_ids=cbl_docs)
+
+    # 4. Starting one-shot replication again to create error based on conflict
+    repl_conflict_config = replicator.configure(source_db=cbl_db,
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: according to step 3 , it should be push/pull replication, but you have used only push. Is this valid?

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+
+    # 5. Validating the event counts and verifying the push and pull event against doc_ids
+    assert sorted(push_docs) == sorted(
+        cbl_doc_ids), "Replication event push docs are not equal to expected no. of docs to be pushed"
+    assert sorted(pull_docs) == sorted(
+        sg_doc_ids), "Replication event pull docs are not equal to expected no. of docs to be pulled"
+    assert len(error_docs) == 0, "Error found in replication events {}".format(error_docs)
+
+
+@pytest.mark.sanity
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: remove sanity here. we need sanity only for one positive flow test

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py [hunk 4]
```diff
@@ -0,0 +1,511 @@
...
+    replicator.removeReplicatorEventListener(repl_conflict, repl_error_change_listener)
+    replicator.stop(repl_conflict)
+    event_dict = get_event_changes(doc_error_repl_event_changes)
+    assert len(event_dict) != 0, "No Event captured. Check the logs for detailed info"
+    for doc_id in event_dict:
+        assert '10409' in event_dict[doc_id]["error_code"], "Conflict error didn't happen. Error Code: {}".format(
+            event_dict[doc_id]["error_code"])
+
+
+# @pytest.mark.sanity
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: remove sanity here

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py [hunk 5]
```diff
@@ -0,0 +1,511 @@
...
+# @pytest.mark.listener
+# @pytest.mark.replication
+# @pytest.mark.parametrize("num_of_docs", [
+#     1000,
+# ])
+# def test_pull_replication_error_event(params_from_base_test_setup, num_of_docs):
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: is this functionality not yet implemented , that is why it is commented?

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py [hunk 6]
```diff
@@ -0,0 +1,511 @@
...
+#     replicator.removeReplicatorEventListener(repl_conflict, repl_error_change_listener)
+#     replicator.stop(repl_conflict)
+#     event_dict = get_event_changes(doc_error_repl_event_changes)
+#     cbl_doc_ids = db.getDocIds(database=cbl_db)
+#     cbl_docs = db.getDocuments(database=cbl_db, ids=cbl_doc_ids)
+#     assert num_of_docs == len(cbl_docs)
+#     assert num_of_docs == len(event_dict)
+
+
+@pytest.mark.sanity
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: remove sanity tag

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py [hunk 7]
```diff
@@ -0,0 +1,511 @@
...
+@pytest.mark.listener
+@pytest.mark.replication
+@pytest.mark.parametrize("num_of_docs", [
+    10,
+    100,
+])
+def test_replication_access_revoke_event(params_from_base_test_setup, num_of_docs):
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: looks like document removal has test something similiar to this. Can I use add few verifications to same test, so that we do not need to duplicate with channel access removal?

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py [hunk 8]
```diff
@@ -0,0 +1,511 @@
...
+    replicator.start(repl)
+    replicator.wait_until_replicator_idle(repl)
+    replicator.removeReplicatorEventListener(repl, repl_change_listener)
+    replicator.stop(repl)
+
+    # 3 changing channel of some docs, so that we get access revoked event
+    sg_docs = sg_client.get_all_docs(url=sg_url, db=sg_db, auth=auth_session)["rows"]
+    docs_to_modify = random.sample(sg_docs, (num_of_docs / 10) * 2)
+    for sg_doc in docs_to_modify:
+        sg_client.update_doc(url=sg_url, db=sg_db, doc_id=sg_doc["id"],
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: to change channel, did you try using some other channel, rather than having blank channel?

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py [hunk 9]
```diff
@@ -0,0 +1,511 @@
...
+            event_dict[doc_id]["flags"] == "AccessRemoved", \
+            'Access Revoked flag is not tagged for document. Flag value: {}'.format(event_dict[doc_id]["flags"])
+
+    # Verifying if the docs, for which access has be revoked, are purged
+    doc_ids = db.getDocIds(cbl_db)
+    for sg_doc in docs_to_modify:
+        assert sg_doc["id"] not in doc_ids, "channel access removal didn't purge the docs from cbl db"
+
+
+@pytest.mark.sanity
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: remove the sanity tag

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py [hunk 10]
```diff
@@ -0,0 +1,511 @@
...
+@pytest.mark.listener
+@pytest.mark.replication
+@pytest.mark.parametrize("num_of_docs", [
+    10,
+    100,
+])
+def test_replication_delete_event(params_from_base_test_setup, num_of_docs):
+    """
+    @summary:
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: This summary looks same as above test. Can you change the summary according to the test?

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py [hunk 11]
```diff
@@ -0,0 +1,511 @@
...
+    assert len(event_dict) != 0, "Replication listener didn't caught events. Check app logs for detailed info"
+    for doc_id in event_dict:
+        assert event_dict[doc_id]["flags"] == "1" or event_dict[doc_id]["flags"] == "[DocumentFlagsDeleted]" or \
+            event_dict[doc_id]["flags"] == "Deleted", \
+            'Deleted flag is not tagged for document. Flag value: {}'.format(event_dict[doc_id]["flags"])
+
+    # Verifying if the docs, for which access has be revoked, are purged
+    doc_ids = db.getDocIds(cbl_db)
+    for sg_doc in docs_to_modify:
+        assert sg_doc["id"] not in doc_ids, "channel access removal didn't purge the docs from cbl db"
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: as it is not channel access, can you modify assert stmt according to the test?

File: testsuites/CBLTester/System_test_multiple_device/test_p2p_replication_eventing.py [hunk 12]
```diff
@@ -0,0 +1,290 @@
...
+
+    # 5. Validating the event counts and verifying the push and pull event against doc_ids
+    assert sorted(push_docs) == sorted(
+        client_doc_ids), "Replication event push docs are not equal to expected no. of docs to be pushed"
+    assert sorted(pull_docs) == sorted(
+        server_doc_ids), "Replication event pull docs are not equal to expected no. of docs to be pulled"
+    assert len(error_docs) == 0, "Error found in replication events {}".format(error_docs)
+
+
+@pytest.mark.sanity
```
Comment (sridevi-15) [inline testsuites/CBLTester/System_test_multiple_device/test_p2p_replication_eventing.py]: you can remove sanity tag.  add replicating event tag to all tests

File: testsuites/CBLTester/System_test_multiple_device/test_p2p_replication_eventing.py [hunk 13]
```diff
@@ -0,0 +1,290 @@
...
+    client_docs = db_obj_client.getBulkDocs(cbl_db_client)
+
+    # verifying that replication doesn't overwrite the docs after conflict
+    for doc in client_docs:
+        if "client_doc" in doc:
+            assert client_docs[doc]["updates-cbl"] == 1, \
+                "Conflict error didn't stop replication of docs properly.\n {}".format(doc)
+
+
+@pytest.mark.sanity
```
Comment (sridevi-15) [inline testsuites/CBLTester/System_test_multiple_device/test_p2p_replication_eventing.py]: u can remove sanity tag

File: testsuites/CBLTester/System_test_multiple_device/test_peer_to_peer.py [hunk 14]
```diff
@@ -50,7 +50,8 @@ def test_peer_to_peer_1to1_valid_values(params_from_base_test_setup, server_setu
         db_obj_client.create_bulk_docs(num_of_docs, "replication", db=cbl_db_client, channels=channels)
 
     # Now set up client
-    repl = peerToPeer_client.client_start(host=server_host, server_db_name=db_name_server, client_database=cbl_db_client, continuous=continuous, replication_type=replicator_type, endPointType=endPointType)
+    repl = peerToPeer_client.configure(host=server_host, server_db_name=db_name_server, client_database=cbl_db_client, continuous=continuous, replication_type=replicator_type, endPointType=endPointType)
```
Comment (sridevi-15) [inline testsuites/CBLTester/System_test_multiple_device/test_peer_to_peer.py]: I hope couple of tests under this file are tested as there are some changes happened on p2p start and stop server

Comment (sridevi-15) [reply testsuites/CBLTester/System_test_multiple_device/test_peer_to_peer.py]: just wondering, is there any need to split the flow into two methods? Instead of using 2 statements, why can we have it in one method? Is there any requirement to have in 2 methods?

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py [hunk 15]
```diff
@@ -0,0 +1,511 @@
...
+from keywords.MobileRestClient import MobileRestClient
+from keywords.utils import log_info
+from keywords.utils import get_event_changes
+from CBLClient.Database import Database
+from CBLClient.Replication import Replication
+from libraries.testkit import cluster
+
+
+@pytest.fixture(scope="function")
+def setup_teardown_test(params_from_base_test_setup):
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: forgot to ask about this. This teardown is not used anywhere,so remove this .

File: CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/server/RequestHandler/ReplicatorRequestHandler.java [hunk 16]
```diff
@@ -73,15 +122,16 @@ public Replicator changeGetReplicator(Args args){
         return change.getStatus();
     }
 
-    public int changeListenerChangesCount(Args args) {
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/server/RequestHandler/ReplicatorRequestHandler.java]: This method is still there in class. Diff is just showing like as this as it has same code snippet

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer.NetCore/Program.cs [hunk 17]
```diff
@@ -50,8 +51,18 @@ static void Main(string[] args)
             Couchbase.Lite.Support.NetDesktop.Activate();
             Extend();
 
-            Couchbase.Lite.Support.NetDesktop.EnableTextLogging("TextLogging");
-            Database.SetLogLevel(Logging.LogDomain.All, Logging.LogLevel.Info);
+            Database.Log.Console.Level = Logging.LogLevel.Debug;
+            Database.Log.Console.Domains = Logging.LogDomain.All;
+            String logPath = Path.Combine(Directory.GetCurrentDirectory(), "LogTestLogs");
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer.NetCore/Program.cs]: The logic is consistent and is enforced based on platform. The current directory format is different across all platform so we can't have same format. Current directory is the path for installed app and hence it will always be there.

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS.xcodeproj/project.pbxproj [hunk 18]
```diff
@@ -22,8 +22,6 @@
 		294FB20C1FEAFD9000DD0C23 /* DatabaseRequestHandler.swift in Sources */ = {isa = PBXBuildFile; fileRef = 294FB20B1FEAFD9000DD0C23 /* DatabaseRequestHandler.swift */; };
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS.xcodeproj/project.pbxproj]: we can add the extension in .gitignore file if you are confident that we don't need it

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/ReplicatorRequestHandler.swift [hunk 19]
```diff
@@ -91,7 +123,7 @@ public class ReplicatorRequestHandler {
     }
 }
 
-class MyReplicationChangeListener : NSObject  {
+public class MyReplicationChangeListener : NSObject  {
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/ReplicatorRequestHandler.swift]: Yes, because we are using same listener in P2P

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/Server.swift [hunk 20]
```diff
@@ -50,7 +50,6 @@ public class Server {
     let memory = Memory()
     
     public init() {
-        Database.setLogLevel(LogLevel.info, domain: LogDomain.all)
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/Server.swift]: Added Database.log.console.level = LogLevel.debug

File: CBLClient/Apps/CBLTestServer-iOS/Frameworks/CouchbaseLiteSwift.framework/Headers/CouchbaseLiteSwift-Swift.h [hunk 21]
```diff
@@ -0,0 +1,186 @@
+// Generated by Apple Swift version 4.2.1 effective-3.4.1 (swiftlang-1000.11.42 clang-1000.11.45.1)
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-iOS/Frameworks/CouchbaseLiteSwift.framework/Headers/CouchbaseLiteSwift-Swift.h]: removed framework

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer.NetCore/TestServer.NetCore.csproj [hunk 22]
```diff
@@ -11,11 +11,10 @@
   </PropertyGroup>
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer.NetCore/TestServer.NetCore.csproj]: added *.csproj to git ignore

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer.iOS/TestServer.iOS.csproj [hunk 23]
```diff
@@ -107,7 +107,7 @@
   </ItemGroup>
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer.iOS/TestServer.iOS.csproj]: added *.csproj to git ignore

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer.UWP/TestServer.UWP.csproj [hunk 24]
```diff
@@ -131,7 +131,7 @@
   </ItemGroup>
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer.UWP/TestServer.UWP.csproj]: added *.csproj to git ignore

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer/TestServer.csproj [hunk 25]
```diff
@@ -15,7 +15,7 @@
   <ItemGroup>
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer/TestServer.csproj]: added *.csproj to git ignore

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS.xcodeproj/xcshareddata/xcschemes/CBLTestServer-iOS-EE.xcscheme [hunk 26]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS.xcodeproj/xcshareddata/xcschemes/CBLTestServer-iOS-EE.xcscheme]: if you can confirm, I'll add this file to git ignore

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS.xcodeproj/xcshareddata/xcschemes/CBLTestServer-iOS.xcscheme [hunk 27]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS.xcodeproj/xcshareddata/xcschemes/CBLTestServer-iOS.xcscheme]: if you can confirm, I'll add this file to git ignore

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/DictionaryRequestHandler.swift [hunk 28]
```diff
@@ -40,13 +40,6 @@ public class DictionaryRequestHandler {
             let data: [String: Any]? = args.get(name: "data")!
 
             return dictionary.setData(data)
-            
-        case "dictionary_setValue":
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/DictionaryRequestHandler.swift]: duplicate

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py (hunk 2 above)
Comment (hrajput89) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: Modified the summary for "push" only

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py (hunk 3 above)
Comment (hrajput89) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: removed

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py (hunk 4 above)
Comment (hrajput89) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: removed

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py (hunk 5 above)
Comment (hrajput89) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: Yes

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py (hunk 6 above)
Comment (hrajput89) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: removed

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py (hunk 7 above)
Comment (hrajput89) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: Please do. You can change it in your branch or make your commit here

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py (hunk 8 above)
Comment (hrajput89) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: Yes, having blank is same as changing other none.

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py (hunk 9 above)
Comment (hrajput89) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: removed

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py (hunk 10 above)
Comment (hrajput89) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: done

File: testsuites/CBLTester/System_test_multiple_device/test_p2p_replication_eventing.py (hunk 12 above)
Comment (hrajput89) [reply testsuites/CBLTester/System_test_multiple_device/test_p2p_replication_eventing.py]: Removed

File: testsuites/CBLTester/System_test_multiple_device/test_p2p_replication_eventing.py (hunk 13 above)
Comment (hrajput89) [reply testsuites/CBLTester/System_test_multiple_device/test_p2p_replication_eventing.py]: removed

File: testsuites/CBLTester/System_test_multiple_device/test_peer_to_peer.py (hunk 14 above)
Comment (hrajput89) [reply testsuites/CBLTester/System_test_multiple_device/test_peer_to_peer.py]: we need to add filters and listener to config. We add them only in rare cases. I've ran all the tests for P2P after modifying API. They all are passing

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py (hunk 15 above)
Comment (hrajput89) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: Removed

File: CBLClient/Apps/CBLTestServer-iOS/Frameworks/CouchbaseLiteSwift.framework/Headers/CBLErrors.h (hunk 1 above)
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-iOS/Frameworks/CouchbaseLiteSwift.framework/Headers/CBLErrors.h]: removed

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py (hunk 11 above)
Comment (hrajput89) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: changed to "Revoked access to channel didn't purge the docs from cbl db"

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/ReplicatorRequestHandler.swift (hunk 19 above)
[the remark this answers is not in the document]
Comment (sridevi-15) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/ReplicatorRequestHandler.swift]: it worked before with p2p without having publie , right?

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py (hunk 5 above)
Comment (sridevi-15) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py]: can we remove this test then?

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS.xcodeproj/xcshareddata/xcschemes/CBLTestServer-iOS-EE.xcscheme (hunk 26 above)
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS.xcodeproj/xcshareddata/xcschemes/CBLTestServer-iOS-EE.xcscheme]: Added *.xcscheme to gitignore

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS.xcodeproj/project.pbxproj (hunk 18 above)
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS.xcodeproj/project.pbxproj]: Added *.pbxproj to gitignore

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS.xcodeproj/xcshareddata/xcschemes/CBLTestServer-iOS.xcscheme (hunk 27 above)
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS.xcodeproj/xcshareddata/xcschemes/CBLTestServer-iOS.xcscheme]: Added *.xcscheme to gitignore
